projects/vdk-plugins/vdk-jupyter/vdk-jupyterlab-extension/vdk-jupyterlab-extension/dict_object.py

```diff
@@ -0,0 +1,14 @@
+# Copyright 2023-2023 VMware, Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+
+class DictObj:
+    def __init__(self, in_dict: dict):
+        assert isinstance(in_dict, dict)
+        for key, val in in_dict.items():
+            if isinstance(val, (list, tuple)):
+                setattr(
+                    self, key, [DictObj(x) if isinstance(x, dict) else x for x in val]
+                )
+            else:
+                setattr(self, key, DictObj(val) if isinstance(val, dict) else val)
```

Comment:
What is the benefit of having a dynamic type based on json properties, in favour of having a class containing an enum of properties that are accompanied by their description? 

I think enum is better for enumerating nomenclatures that are static during runtime. It is a subject to compiler checks. For example, a duplication could be prevented.